### DISCUSSION TITLE
[Fixed on 0.7.1] Unexpected error using spline agent 0.7 with AWS Glue 3.0

### DISCUSSION THREAD

**Discussion by araujo-filipe** [AbsaOSS/spline-spark-agent#345] [Q&A] [answered]
Hi guys, 

I've been using a custom version of the spline agent for collecting data lineage from AWS Glue spark jobs.
Now AWS Glue released a new version that uses spark 3.1 so I decided to try to update the spline agent to 0.7.0, which supports it.

I'm getting an 
`ERROR SplineQueryExecutionListener: Unexpected error occurred during lineage processing for application ...`
and after digging a bit I could retrieve a better error message:
`java.lang.NumberFormatException: For input string: "op-0" `

After looking at the code, this seems to be related with the idGenerators created on the latest versions of the spline agent, but unfortunately my knowledge of scala isn't sufficient to understand why this is breaking the spline agent.

Could someone give a hint on how to solve or further debug this issue?

Thanks in advance!

Edit: replacing "spline" with "spline agent" since that's the only component used to get this error.

**Comment by wajda** [comment]
Interesting. What Spline server version are you using? 
In agent 0.7 we changed the ID format for some entities, particularly the operation ID has changed from a "$number" to "op-$number". But the change should be irrelevant for the server, as the server treats those as just string anyway.

**Reply by wajda** [reply]
I just thought the exception might be thrown during downgrading the lineage format to an older version (it happens when a newer agent is talking to an older server - version 0.5 if I'm not mistaken or even less). This certainty a bug, we'll fix it, but I'd encourage you to upgrade all Spline components to the latest versions for the best experience.

**Reply by araujo-filipe** [reply]
Hey @wajda, thanks for the quick response as always :)
Maybe my wording wasn't the best (I'll edit the post), but we are only taking advantage of the spline agent, we do not use the spline server. Also, the error occurs before sending the request with the execution plan, so it is not yet interacting with any other component.

**Reply by wajda** [reply]
It would be much easier if I could see the full error stack trace. But from what I can guess the error is happening on attempt to parse "op-1" as an integer. There is the only place in the agent where it is happening - the conversion from the current lineage format to the older one. That in turn happens when the http dispatcher wasn't able to recognize the version of the Spline Server and assumes it's talking to the oldest supported one.

https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/dispatcher/httpdispatcher/modelmapper/ModelMapperV1.scala

This could be the case if you don't use Spline Server, but you are using an HTTP Dispatcher - you'll always get output in the oldest JSON format.

A temporary fix (until we make a proper one) is to change the operation ID pattern from `"op-{0}"` to just `"{0}"`

https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/IdGenerators.scala

**Reply by wajda** [reply]
added a ticket (#346) and a permanent fix (#347) that will be released in version 0.7.1

**Reply by wajda** [reply]
Version 0.7.1 is out and the issue is fixed.

**Reply by araujo-filipe** [reply]
Works like a charm, thanks once again for the amazing spline agent 👍

**Comment by araujo-filipe** [comment, marked as answer]
Fixed on version 0.7.1
